app/components/HeaderNotifications/index.tsx

```diff
@@ -103,21 +104,27 @@ const NotificationsDropdown = () => {
   );
 
   const { unreadCount } = notificationsData;
+
   return (
     <Dropdown
       show={notificationsOpen}
       toggle={() => {
-        setNotificationsOpen(!notificationsOpen);
+        setNotificationsOpen((prev) => !prev);
+
         if (!notificationsOpen) {
           dispatch(fetchNotificationFeed());
         } else {
+          // Optimistic mark all as read first to avoid flickering
+          dispatch({ type: NotificationsFeed.MARK_ALL.SUCCESS });
           dispatch(markAllNotifications());
```

Comment:
Mark as completed on `NotificationsFeed.MARK_ALL.BEGIN` in the reducer instead of dispatching wrong actions